code/modules/games/cards.dm

```diff
@@ -0,0 +1,443 @@
+/datum/playingcard
+	var/name = "playing card"
+	var/card_icon = "card_back"
+	var/back_icon = "card_back"
+
+/obj/item/weapon/deck
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/playing_cards.dmi'
+	var/list/cards = list()
+	var/cooldown = 0 // to prevent spam shuffle
+
+/obj/item/weapon/deck/holder
+	name = "card box"
+	desc = "A small leather case to show how classy you are compared to everyone else."
+	icon_state = "card_holder"
+
+/obj/item/weapon/deck/cards
+	name = "deck of cards"
+	desc = "A simple deck of playing cards."
+	icon_state = "deck"
+
+/obj/item/weapon/deck/cards/New()
+	..()
+	var/datum/playingcard/P
+	for(var/suit in list("spades","clubs","diamonds","hearts"))
+
+		var/colour
+		if(suit == "spades" || suit == "clubs")
+			colour = "black_"
+		else
+			colour = "red_"
+
+		for(var/number in list("ace","two","three","four","five","six","seven","eight","nine","ten"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]num"
+			P.back_icon = "card_back"
+			cards += P
+
+		for(var/number in list("jack","queen","king"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]col"
+			P.back_icon = "card_back"
+			cards += P
+
+	for(var/i = 0, i<2, i++)
+		P = new()
+		P.name = "joker"
+		P.card_icon = "joker"
+		cards += P
+
+/obj/item/weapon/deck/attackby(obj/O as obj, mob/user as mob)
+	if(istype(O,/obj/item/weapon/hand))
+		var/obj/item/weapon/hand/H = O
+		if(H.parentdeck == src)
+			for(var/datum/playingcard/P in H.cards)
+				cards += P
+			qdel(H)
+			to_chat(user,"<span class='notice'>You place your cards on the bottom of [src]</span>.")
+			return
+		else
+			to_chat(user,"<span class='warning'>You can't mix cards from other deck!</span>")
```

Comment:
`You can't mix cards from other deck` should be either `You can't mix cards from different decks` or `You can't add cards from another deck`